File: app_flutter/app_cua_toi/lib/data/models/api_response.dart
class ApiResponse<T> {
  int code;
  String message;
  T? result;

  ApiResponse({this.code = 1000, this.message = "Success!", this.result});

  factory ApiResponse.fromJson(
    Map<String, dynamic> json,
    T Function(dynamic)? fromJsonT,
  ) {
    return ApiResponse(
      code: json['code'] ?? 1000,
      message: json['message'] ?? '',
      result: json['result'] != null && fromJsonT != null
          ? fromJsonT(json['result'])
          : null,
    );
  }
}

File: giao_dien_quan_ly/webapp/src/api/types/api.response.ts
export interface ApiResponse<T> {
  code?: number;
  message?: string;
  result?: T;
}

// Cấu trúc Page khớp với JSON: "page": { "size": 10, ... }
export interface PageInfo {
  size: number;
  number: number;
  totalElements: number;
  totalPages: number;
}

export interface PageResponse<T> {
  content: T[];
  page: PageInfo; 
}